sphinx/events.py

```diff
@@ -100,11 +100,16 @@ def emit(self, name: str, *args: Any) -> List:
         results = []
         listeners = sorted(self.listeners[name], key=attrgetter("priority"))
         for listener in listeners:
-            if self.app is None:
-                # for compatibility; RemovedInSphinx40Warning
-                results.append(listener.handler(*args))
-            else:
-                results.append(listener.handler(self.app, *args))
+            try:
+                if self.app is None:
+                    # for compatibility; RemovedInSphinx40Warning
+                    results.append(listener.handler(*args))
+                else:
+                    results.append(listener.handler(self.app, *args))
+            except SphinxError:
+                raise
+            except Exception as exc:
+                raise ExtensionError(exc)
```

Comment:
```suggestion
                raise ExtensionError('Handler {} for event {} threw an exception'.format(listener, event)) from exc
```
That way the chance is slightly higher that the user can recover the original stack trace.

Reply:
Thanks!

Reply:
Any reason for using `%` formatting instead of `.format`? Just so I can pick the right one for future contributions.

Reply:
Personally, I prefer it to `.format`. I think both are right.